src/components/structures/RightPanel.js

```diff
@@ -205,7 +216,10 @@ module.exports = React.createClass({
     }, 500),
 
     onAction: function(payload) {
-        if (payload.action === "view_user") {
+        if (payload.action === "event_notification") {
+            // Try and re-caclulate any badge counts we might have
+            this.forceUpdate();
```

Comment:
Will you not need an equivalent for when the user reads the notification and the count goes down again?

Reply:
You would think so, but something somewhere magically makes it work. I tested in Chrome and Firefox and for some reason it happily goes down but doesn't magically go up.

Reply:
yay magic :/